### PR TITLE
Wad 25 node infrastructure

### PR DESCRIPTION
**Pull request by lammia07**
Create NodeJs infrastructure:
- express server
- config
- mysql
- routing

### CODE REVIEW

File: features/dbConnection.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+const mysql = require('mysql');
+const config = require('config');
+//...
```
Comment (m4rien0) [inline features/dbConnection.js]: remove?

File: features/dbConnection.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+const poolConfig = config.get('MySqlConnectionPool');
+
+// Ist keine Klasse da wir private fields haben wollen. 
+// In unserem Fall 'connection' und 'handleError'
```
Comment (m4rien0) [inline features/dbConnection.js]: Change comment to english.

File: features/test/testRouter.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+const testService = require('./testService')
+
+// Erstellen von 2 router
+// Wir benötigen 2 router da wir die routen
+// - /projects
+// - /api/projects
+// ansteuern möchten.
+// Da die präfixe '/projects' und '/api/projects' erst beim Hinzufügen
+// dieses routers gesetzt werden, (wir wollen hier keinen doppelten Code schreiben)
+// müssen hier 2 Router definieren.
```
Comment (m4rien0) [inline features/test/testRouter.js]: Change comment to English.

File: features/test/testRouter.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+const viewRouter = express.Router();
+const apiRouter = express.Router();
+
+
+// Definieren der routen für die view files. In unserem fall sind dies plain HTML files
```
Comment (m4rien0) [inline features/test/testRouter.js]: See previous comment.

File: features/test/testRouter.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+viewRouter.get('/', (req, res) => {
+    res.sendFile(path.join(__dirname, 'test.html'));
+});
+
+// Definieren der routen für die api. Die Api returniert nur Daten, keine Oberfläche. 
+// In unserem fall im JSON Format
```
Comment (m4rien0) [inline features/test/testRouter.js]: See previous comment.

File: features/test/testService.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
+const dbConnection = require('../dbConnection');
+
+// Erstellen einer Service Klasse, die sich um CRUD Aktionen mit der Datenbank kümmert
+// C - Create
+// R - Read
+// U - Update
+// D - Delete
+// Achtung in JS gibt es keinen private deklarator bei Klassen
```
Comment (m4rien0) [inline features/test/testService.js]: Change comment to English.

File: features/test/testService.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+class TestService {
+    constructor() {
+    }
+
+    get(succes) {
+        dbConnection.select("Select * FROM tests", succes);
```
Comment (m4rien0) [inline features/test/testService.js]: Would do all caps for SQL queries.
```suggestion
        dbConnection.select("SELECT * FROM tests", success);
```

File: features/test/testService.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+// C - Create
+// R - Read
+// U - Update
+// D - Delete
+// Achtung in JS gibt es keinen private deklarator bei Klassen
+class TestService {
+    constructor() {
+    }
+
+    get(succes) {
```
Comment (m4rien0) [inline features/test/testService.js]:
```suggestion
    get(success) {
```

File: features/test/testService.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    }
+    
+    insert(test) {
+        this.tests[this.tests.length] = test;
+        return test;
+    }
+  };
+
+// Testservice wird als Singleton deklariert. 
+// Bedeutet es kann nur 1 Instanz des Service geben
```
Comment (m4rien0) [inline features/test/testService.js]: Change comment to English.

File: package.json [hunk 10]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "webapp-ws20-projects",
+  "version": "1.0.0",
```
Comment (m4rien0) [inline package.json]:
```suggestion
  "version": "1.2.0",
```

File: package.json [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+  "description": "web application development projekt of the group SnickersVsMars for the FH-JOANNEUM",
```
Comment (m4rien0) [inline package.json]:
```suggestion
  "description": "Education web application development project of the group SnickersVsMars for the FH-JOANNEUM",
```

File: package.json [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/SnickersVsMars/webapp-ws20-projects.git"
+  },
+  "keywords": [
+    "nodejs",
+    "bootstrap"
+  ],
+  "author": "Michael Lamprecht",
```
Comment (m4rien0) [inline package.json]:
```suggestion
  "author": "SnickersVsMars",
```

File: server.js [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+//Express error handling middleware
+server.use((request,response)=>{
+   response.type('text/plain');
+   response.status(505);
+   response.send('Error page');
+});
+
+//Binding to a port
+server.listen(port, () => {
+  console.log('Express server started at port 3000');
```
Comment (m4rien0) [inline server.js]:
```suggestion
  console.log('Express server started on port ' + port);
```

Comment (m4rien0) [reply server.js]: Port should be from config file here as well (not hard coded).

**Review by m4rien0 — CHANGES_REQUESTED** [review]

File: features/dbConnection.js (hunk 1 above)
Comment (lammia07) [reply features/dbConnection.js]: removed

File: features/dbConnection.js (hunk 2 above)
Comment (lammia07) [reply features/dbConnection.js]: translated

File: features/test/testRouter.js (hunk 3 above)
Comment (lammia07) [reply features/test/testRouter.js]: translated

File: features/test/testRouter.js (hunk 4 above)
Comment (lammia07) [reply features/test/testRouter.js]: translated

File: features/test/testRouter.js (hunk 5 above)
Comment (lammia07) [reply features/test/testRouter.js]: translated

File: features/test/testService.js (hunk 9 above)
Comment (lammia07) [reply features/test/testService.js]: translated

**Review by m4rien0 — COMMENTED** [review]
Seems good :)

**Pull request merged**